[docs] Avoid confusion between layout grid and data grid (#22681)
mui-org/material-ui: docs/src/pages/components/grid/grid.md

{{"component": "modules/components/ComponentLinkHeader.js"}}

> > ⚠️ The `Grid` component shouldn't be confused with a data grid. The component is closer to a layout grid. Head to this page for [a `DataGrid` component](/components/data-grid/).

Reviewer: `this page` is similar to `here` links. How about

```suggestion
> ⚠️ The `Grid` component shouldn't be confused with a data grid. The component is closer to a layout grid. For a data grid head to [our `DataGrid` component](/components/data-grid/).
```

> > ⚠️ The `Grid` component shouldn't be confused with a data grid. The component is closer to a layout grid. For a data grid head to [our `DataGrid` component](/components/data-grid/).

Reviewer 2: data-grid?

Author: I'm not aware of any prior usages of the hyphenated version in our docs. What's the advantage?

Reviewer 3: ```suggestion
> ⚠️ The `Grid` component shouldn't be confused with a data grid; it is closer to a layout grid. For a data grid head to [the `DataGrid` component](/components/data-grid/).
```
